core/src/main/java/greencity/validators/AddressValidator.java

```diff
@@ -0,0 +1,75 @@
+package greencity.validators;
+
+import com.google.maps.model.AddressComponent;
+import com.google.maps.model.AddressComponentType;
+import com.google.maps.model.GeocodingResult;
+import com.google.maps.model.LatLng;
+import greencity.annotations.ValidAddress;
+import greencity.dto.CreateAddressRequestDto;
+import greencity.dto.google.AddressResponseFromGoogleAPI;
+import greencity.dto.location.CoordinatesDto;
+import greencity.service.google.GoogleApiService;
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+import java.util.Arrays;
+import lombok.RequiredArgsConstructor;
+
+@RequiredArgsConstructor
+public class AddressValidator implements ConstraintValidator<ValidAddress, CreateAddressRequestDto> {
+    private final GoogleApiService googleApiService;
+    private static final double DELTA = 0.007;
+    private static final int LANGUAGE_CODE_FOR_UA = 0;
+
+    @Override
+    public void initialize(ValidAddress constraintAnnotation) {
+        //
+    }
+
+    @Override
+    public boolean isValid(CreateAddressRequestDto createAddressRequestDto, ConstraintValidatorContext context) {
+        String placeId = createAddressRequestDto.getPlaceId();
+        CoordinatesDto coordinates = createAddressRequestDto.getCoordinates();
+        LatLng latLng = new LatLng(coordinates.getLatitude(), coordinates.getLongitude());
+
+        GeocodingResult geoResult = googleApiService.getResultFromGeoCode(placeId, LANGUAGE_CODE_FOR_UA);
```

Comment:
Can we use some in-memory cache?
If the user does several call with the same address we do not need to call Google API several times